pkg/runs/runsDownload.go

```diff
@@ -0,0 +1,188 @@
+func GetArtifactPathsFromRestApi(runId string, apiServerUrl string) ([]string, error) {
+
+	var err error = nil
+	var artifactPaths []string
+	log.Println("Retrieving artifact paths for the given run")
+
+	// An HTTP client which can communicate with the api server in an ecosystem.
+	restClient := api.InitialiseAPI(apiServerUrl)
+
+	var artifactsList []galasaapi.ArtifactIndexEntry
+	artifactsList, httpResponse, err := restClient.ResultArchiveStoreAPIApi.
+		GetRasRunArtifactList(context.Background(), runId).
+		Execute()
+
+	if err != nil {
+		err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_RETRIEVING_ARTIFACTS_FAILED, err.Error())
+	} else {
+		for _, artifact := range artifactsList {
+			artifactPaths = append(artifactPaths, artifact.GetPath())
+		}
+	}
+	defer httpResponse.Body.Close()
+	log.Printf("%v artifact path(s) found", len(artifactPaths))
+
+	return artifactPaths, err
+}
+
+// Writes an artifact to the host's file system, creating a new directory for the run's artifacts
+// to be written to. Existing files are only overwritten if the --force option is used as part of
+// the "runs download" command.
+func WriteArtifactToFileSystem(
+	fileSystem utils.FileSystem,
+	runDirectory string,
+	artifactPath string,
+	fileDownloaded *os.File,
+	shouldOverwrite bool) error {
+
+	var err error = nil
+	bufferCapacity := 1024
+
+	pathParts := strings.Split(artifactPath, "/")
+	fileName := pathParts[len(pathParts) - 1]
+	targetFilePath := filepath.Join(runDirectory, fileName)
+
+	// Check if a new file should be created or if an existing one should be overwritten.
+	fileExists, err := fileSystem.Exists(targetFilePath)
+	if err == nil {
+		if fileExists && !shouldOverwrite {
+
+			// The --force flag was not provided, throw an error.
+			err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_CANNOT_OVERWRITE_FILE, targetFilePath)
+			return err
+		}
+
+		// Set up the directory structure and artifact file on the host's file system
+		var newFile io.Writer = nil
+		newFile, err = CreateEmptyArtifactFile(fileSystem, targetFilePath)
+		if err == nil {
+			log.Printf("Writing artifact '%s' to '%s' on local file system", fileName, targetFilePath)
+
+			// Set up a byte buffer to gradually read the downloaded file to avoid out-of-memory issues.
+			reader := bufio.NewReader(fileDownloaded)
+			buffer := make([]byte, bufferCapacity)
+
+			// Read a chunk of the downloaded file into the buffer and write the buffer's contents to the
+			// newly-created file.
+			for {
+				bytesRead, err := reader.Read(buffer)
+				if err != nil {
+					if err == io.EOF {
+						// There was nothing else to read.
+						log.Printf("Artifact '%s' written to '%s' OK", fileName, targetFilePath)
+						break
+					}
+					return err
+				}
+
+				_, err = newFile.Write(buffer[:bytesRead])
+				if err != nil {
+					err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_FAILED_TO_WRITE_FILE, targetFilePath, err.Error())
+					return err
```

Comment:
Not needed. Dropping to the bottom returns the error anyway,.

Reply:
Addressed in https://github.com/galasa-dev/cli/pull/92/commits/ff21f567cdc46463eafc4316afd8bb72ed7bcfdf